### ISSUE TITLE
Czy jest w planach integracja z innymi aplikacjami tego typu?

### ISSUE THREAD

**Issue by MadryPan1987** [ProteGO-Safe/specs#218]
Cześć,
Jak w temacie. Czy jest w planach integracja z aplikacjami w niemczech, francji hiszpani, włoszech i uk?
Często podróżuje w tamte rejony i fajnie jakby protego działało przynajmniej w europie ;p

**Comment by SeraMoon** [comment]
Przecież jest zintegrowana. Mądry Panie, przeczytaj na czym opiera się praca tych aplikacji. Wszystkie te aplikacje (corona-warn, protego-safe) są **nakładkami** na Exposure Notification od Google + Apple. To nie są prawdziwe odrębne aplikacje, ale potężna część zawarta jest w API Gapple (tak mówimy o zmowie Apple + Google). 

Na razie pilniejszym tematem jest #213 (Skarga do Europe Direct oraz państwa EU powinny zebrać się do kupy i zażądać otwarcia kodu EN).

**Comment by MateuszRomanow** [comment]
> Cześć,
> Jak w temacie. Czy jest w planach integracja z aplikacjami w niemczech, francji hiszpani, włoszech i uk?
> Często podróżuje w tamte rejony i fajnie jakby protego działało przynajmniej w europie ;p

To tak zwana Interoperacyjności, czyli zdolność aplikacjo do współpracy z innymi aplikacjami w UE. Obecnie wszystkie kraje w ramach UE w tym PL bardzo intensywnie pracują nad wdrożeniem standardu interoperacyjności. Całość musi się ze sobą komunikować po stronie zaplecza (backendu) gdzie odbywa się przesyłanie kluczy. Dużym wyzwaniem jest:
- ustalenie w jaki sposób turysta, który pojechał np z DE do IT a prześle swoje klucze, powinien je przesłać do IT i wszystkich innych turystów którzy byli w tym czasie w IT a nie np do Obywateli z PL. 
- konfiguracja przepływu po stronie zaplecza
- dostarczanie kluczy w taki sposób, żeby nie dociążyć transferu danych użytkowników
Wszystkie te tematy są dynamicznie rozwijane na linii eHealth Network, EU<>Google, EU<>Apple, PL<>eHealth, PL<>Google, PL<>Apple. Ponieważ tworzy się standard tego jak ten proces ma wyglądać, zespołu programistów autorów wszystkich aplikacji w UE biorą w tym udział, w tym także nasz zespół od ProteGO Safe.

**Comment by MateuszRomanow** [comment]
> Ktoś kto miał kontakt z użytkownikiem ProteGO-Safe i ten użytkownik był zakażony SARS-CoV-2 (dodatni test) - dostanie powiadomienie o potrzebie zgłoszenia się na test też przez aplikację Corona-Warn.

Nie jest to prawdą. W tym momencie tak to nie działa. Zacznie działać, jak zostanie wdrożona interoperacyjność, nad którą pracują teraz wszystkie kraje UE. My blisko pracujemy właśnie z Niemcami w sprawie interoperacyjności PL<>DE ze względu na dużą wymianę Obywateli. Co ciekawe, na początku rozmów (4-6 tygodni temu) temat był zapowiadany jako stosunkowo łatwy, natomiast w minionym tygodniu zostało przeniesione spotkanie programistów eHealth Network z powodu zbyt powolnych postępów s pracach. Jakaś beta rozwiązania powinna być gotowa w ciągu 4-8 tygodni. Pracuje nad tym teraz w sumie, około 1000 developerów ze wszystkich krajów UE - w tym nasi programiści od ProteGO Safe. Krajów, które mają lub planują wdrożenie aplikacji (a planują wszystkie kraje UE, my byliśmy jednym z pierwszych).

**Comment by pkleczko** [comment]
@MadryPan1987 tak jak @MateuszRomanow wspomniał - tak są takie plany i są podjęte odpowiednie działania na szczeblu EU. Będziemy informować o postępach, póki co temat jest jeszcze na etapie konsultacji. Wdrożenie to kwestia kilkunastu tygodni/kilku miesięcy.

Czy możemy zamknąć to issue?

**Comment by MateuszRomanow** [comment]
@pkleczko proponuję zostawić to Issue ponieważ temat będzie rozwojowy. Będziemy tu aktualizować postępy.

**Comment by pkleczko** [comment]
[Tutaj](https://github.com/eu-federation-gateway-service/efgs-federation-gateway) znajduje się projekt EU Federation Gateway Service (EFGS), czyli rozwiązania wypracowanego na forum europejskim jako serwis pozwalający na wymianę kluczy pomiędzy aplikacjami działającymi w różnych krajach. Projekt jest wciąż konsultowany i dopracowywane są szczegóły zarówno techniczne jak i te związane z data privacy i legal. Aktualnie rozpoczyna się faza 'pilotażu', czyli wczesnych testów dla krajów już posiadających swoje aplikacje z modułem EN, w którym bierzemy udział.

Będę informował o dalszych postępach.

**Comment by potiuk** [comment]
Super. Cieszę się że społeczność wokół EN się konkretyzuje i idzie dokładnie w kierunkach w którym powinien. Jeszcze parę rzeczy i już ten projekt powoli zacznie nabierać sensu.  Z tego co widzę to strona techniczna już jest praktycznie zrobiona (jak pisałem w dość prosta). Dużo trudniejsze jest właśnie to dogadanie się na poziomie międzyrządowym jak pisze @pkleczko. Cieszę się że nasze państwo i Ministerstwo zajmuje się właśnie tym (ustalaniem na poziomie prawnym i jak być częścią całego systemu) a nie tworzeniem technologii żeby być "pierwsi" . 

To są bardzo dobre wiadomości. Szkoda że taki kierunek działań jest dopiero teraz po wyrzuceniu mnóstwa publicznych pieniędzy na wcześniejsze próby, które nie miały do końca sensu a były mocno polityczne.

Niemniej, cieszy to, że idziemy w dobrym kierunku. Dzięki za informacje - mam nadzieję że dalej będziecie informować społeczność w miarę regularnie o postępach i nie będzie się trzeba uciekać do konrowersyjnych postów.

**Comment by potiuk** [comment]
No i jeszcze news z dziś. To już jest przedostatni mój postulat - jeszcze tylko niezależna organizacja do nadzoru i będzie naprawdę dobrze:

https://www.technologyreview.com/2020/09/02/1007921/apple-and-google-have-launched-coronavirus-exposure-notifications-without-an-app/

Naprawdę - EN zaczyna być coraz bliżej do bycia faktycznie dobrym rozwiązaniem.

**Comment by kamil-gwozdz** [comment]
@MateuszRomanow Są prowadzone jakieś prace nad tym tematem po polskiej stronie? Irlandia, Niemcy i Włochy już się razem zintegrowali kilka dni temu. Wiadomo kiedy my do nich dołączymy?

**Comment by potiuk** [comment]
No. Przydało by się wiedzieć. Ostatnio jak mnie dziennikarz pytał to nie wiedziałem co odpowiedzieć i czemu - mimo wcześniejszych zapowiedzi że Polska chce i jak najbardziej, na razie nie ma o tym mowy?

**Comment by pkleczko** [comment]
będziemy dołączali w listopadzie do EFGS, prawdopodobnie w okolicach połowy miesiąca.

**Comment by kamil-gwozdz** [comment]
> będziemy dołączali w listopadzie do EFGS, prawdopodobnie w okolicach połowy miesiąca.

@pkleczko Czy to jest planowana data rozpoczęcia prac czy opublikowania wersji wspierającej EFGS?

**Comment by pkleczko** [comment]
Planowany release, prawdopodobnie 16-20.11
